docs/vscode-setup.md

```diff
@@ -0,0 +1,124 @@
+# VS Code Setup
+
+VS Code is an excellent and popular development environment that you can use
+for developing Bristlemouth firmware. Here are some setup instructions.
+
+*Please follow the instructions in
+[`ENV_SETUP.md`](https://github.com/bristlemouth/bm_protocol/blob/develop/ENV_SETUP.md)
+and
+[`BMDK_README.md`](https://github.com/bristlemouth/bm_protocol/blob/develop/src/apps/bm_devkit/BMDK_README.md)
+first.*
+The VS Code setup won't work until you can successfully work in the command line.
+
+At the top level of the repo, if it doesn't exist yet, create a `.vscode` directory.
+It's ignored by git because it contains settings specific to your machine.
+We're going to create and edit files in there.
+
+
+## 1. CMake and the conda env
+
+- Install the CMake Tools extension from Microsoft.
+- If you don't yet have the `code` command in your terminal PATH to open the VS Code app, then:
+    - In VS Code, hit Cmd-Shift-P and choose "Shell Command: Install 'code' command in PATH"
+    - In order to get VS Code to work in your conda environment, you'll need to **always open it from a terminal where you're already in the conda environment**. For example, open terminal, cd to the `bm_protocol` directory, `conda activate bristlemouth`, then `code .` (with the dot argument) meaning "open the current directory in VS Code". Go ahead and quit VS Code and open it this way now.
+
```

Comment:
have you thought about making a bash script that will do this for you? Then you can make a little icon that lives on the task bar. I've done this for sourcetree(git gui app that didn't work properly until I executed it from terminal).

Reply:
Ah interesting! For me personally, typing `code .` in the terminal works really well, but that could be a super  helpful tip for many people Want to add something to this PR @victorsowa12?

Reply:
I would need to do a little test to make sure it works first. I will hold off on adding anything so this PR can go out!